src/Client/Core/PurgeResult.cs

```diff
@@ -18,9 +18,29 @@ public PurgeResult(int count)
         this.PurgedInstanceCount = count;
     }
 
+    /// <summary>
+    /// Initializes a new instance of the <see cref="PurgeResult" /> class.
+    /// </summary>
+    /// <param name="count">The number of instances deleted.</param>
+    /// <param name="isComplete">A value indicating whether the purge operation is complete. 
+    /// If true, the purge operation is complete. All instances were purged.
+    /// If false, not all instances were purged. Please purge again.
+    /// If null, default to legacy purge behavior. All instances were purged.</param>
+    public PurgeResult(int count, bool? isComplete)
+        : this(count)
+    {
+        this.IsComplete = isComplete;
+    }
+
     /// <summary>
     /// Gets the number of purged instances.
     /// </summary>
     /// <value>The number of purged instances.</value>
     public int PurgedInstanceCount { get; }
+
+    /// <summary>
+    /// Gets a value indicating whether the purge operation is complete.
+    /// </summary>
+    /// <value>A value indicating whether the purge operation is complete.</value>
+    public bool? IsComplete { get; }
```

Comment:
The comments indicate you are using `null` to indicate legacy behavior. Considering this is the same as `true`, do we need `null`? Or can we default this to `true`?

Reply:
good question, i think both makes sense, null preserve the semantic of old and new versions better but adds more complexity for client to handle, @cgillum for feedback on this

Reply:
See my comment on the corresponding DTFx PR. It's best to document null as being "undefined" in terms of whether all instances got deleted since it depends on the implementation of the backend (for example, it's true for Azure Storage, but not necessarily for MSSQL).